backend/typescript/services/implementations/campService.ts

```diff
@@ -102,36 +102,43 @@ class CampService implements ICampService {
   }
 
   async updateCampById(campId: string, camp: UpdateCampDTO): Promise<CampDTO> {
-    const oldCamp: Camp | null = await MgCamp.findById(campId);
+    let oldCamp: Camp | null;
+    try {
+      oldCamp = await MgCamp.findById(campId);
 
-    if (!oldCamp) {
-      throw new Error(`Camp' with campId ${campId} not found.`);
-    }
+      if (!oldCamp) {
+        throw new Error(`Camp' with campId ${campId} not found.`);
+      }
 
-    if (oldCamp.active && camp.fee) {
-      throw new Error(`Cannot update fee of active camp`);
-    }
+      if (oldCamp.active && camp.fee) {
+        throw new Error(`Error - cannot update fee of active camp`);
+      }
 
-    try {
-      await MgCamp.findByIdAndUpdate(campId, {
-        $set: {
-          name: camp.name,
-          active: camp.active,
-          ageLower: camp.ageLower,
-          ageUpper: camp.ageUpper,
-          campCoordinators: camp.campCoordinators,
-          campCounsellors: camp.campCounsellors,
-          description: camp.description,
-          earlyDropOff: camp.earlyDropOff,
-          latePickup: camp.latePickup,
-          location: camp.location,
-          startTime: camp.startTime,
-          endTime: camp.endTime,
-          volunteers: camp.volunteers,
-          fee: camp.fee,
-          volunteers: camp.volunteers,
-        },
-      });
+      try {
```

Comment:
I think you can remove this try/catch since the error will be caught by the one wrapping this